### PR TITLE
TBDGen: subscripts, default argument, @objc protocols, vtable stubs, global variables

### PR DESCRIPTION
**Pull request by huonw**

### CODE REVIEW

**Comment by huonw** [comment]
@swift-ci Please smoke test

File: lib/IRGen/GenDecl.cpp [hunk 1]
```diff
@@ -973,6 +973,10 @@ void IRGenModule::emitVTableStubs() {
     // For each eliminated method symbol create an alias to the stub.
     auto *alias = llvm::GlobalAlias::create(llvm::GlobalValue::ExternalLinkage,
                                             F.getName(), stub);
+
+    if (F.getEffectiveSymbolLinkage() == SILLinkage::Hidden)
```
Comment (slavapestov) [inline lib/IRGen/GenDecl.cpp]: What if it's private?

Comment (huonw) [inline lib/IRGen/GenDecl.cpp]: There's a guard above (`isExternallyUsedSymbol`).

Comment (slavapestov) [inline lib/IRGen/GenDecl.cpp]: Ah, gotcha.

**Comment by huonw** [comment]
@swift-ci Please test and merge

**Pull request merged**